frontend/server/src/SecurityTools.php

```diff
@@ -237,4 +242,64 @@ public static function getGitserverAuthorizationToken(
             ]);
         return $token->toString();
     }
+
+    /**
+     * Gets a Bearer authorization token for a particular user to be used the
+     * gitserver that is valid for a single problem for 5 minutes.
+     *
+     * @param array<string, string> $claims
+     * @param string $subject
+     * @return string The Bearer authorization token.
+     */
+    public static function getCourseCloneAuthorizationToken(
+        array $claims,
+        string $subject
+    ): string {
+        // Given that we already have an autoload configured, we cannot use
+        // sodium_compat's (fast) autoloader. Instead, simulate what it does
+        // here, with the full path of the standard autoload file.
+        require_once 'libs/third_party/sodium_compat/autoload.php';
+        \ParagonIE_Sodium_Compat::$fastMult = true;
+
+        require_once 'libs/third_party/constant_time_encoding/src/EncoderInterface.php';
+        require_once 'libs/third_party/constant_time_encoding/src/Base64.php';
+        require_once 'libs/third_party/constant_time_encoding/src/Base64UrlSafe.php';
+        require_once 'libs/third_party/constant_time_encoding/src/Binary.php';
+
+        require_once 'libs/third_party/paseto/src/KeyInterface.php';
+        require_once 'libs/third_party/paseto/src/SendingKey.php';
+        require_once 'libs/third_party/paseto/src/ReceivingKey.php';
+        require_once 'libs/third_party/paseto/src/Keys/SymmetricKey.php';
+        require_once 'libs/third_party/paseto/src/Keys/AsymmetricSecretKey.php';
+        require_once 'libs/third_party/paseto/src/Keys/AsymmetricPublicKey.php';
+        require_once 'libs/third_party/paseto/src/ProtocolCollection.php';
+        require_once 'libs/third_party/paseto/src/ProtocolInterface.php';
+        require_once 'libs/third_party/paseto/src/Protocol/Version1.php';
+        require_once 'libs/third_party/paseto/src/Protocol/Version2.php';
+        require_once 'libs/third_party/paseto/src/Traits/RegisteredClaims.php';
+        require_once 'libs/third_party/paseto/src/JsonToken.php';
+        require_once 'libs/third_party/paseto/src/Purpose.php';
+        require_once 'libs/third_party/paseto/src/Builder.php';
+        require_once 'libs/third_party/paseto/src/Util.php';
+        require_once 'libs/third_party/paseto/src/Parsing/Header.php';
+        require_once 'libs/third_party/paseto/src/Parsing/PasetoMessage.php';
+        require_once 'libs/third_party/paseto/src/Exception/PasetoException.php';
+
+        if (is_null(self::$_courseCloneSecretKey)) {
+            self::$_courseCloneSecretKey = new \ParagonIE\Paseto\Keys\SymmetricKey(
+                base64_decode(OMEGAUP_COURSE_CLONE_SECRET_KEY) . '='
+            );
```

Comment:
```suggestion
            self::$_courseCloneSecretKey = \ParagonIE\Paseto\Keys\SymmetricKey::fromEncodedString(
                OMEGAUP_COURSE_CLONE_SECRET_KEY
            );
```

poner el `=` afuera de `base64_decode` lo que hace es agregar un `=` literal a la llave, eliminando 8 bits de entropía y haciendo que la llave sea innecesariamente débil :S

Reply:
Listo. Ahora ya funciona todo correctamente.